### ISSUE TITLE
Error: GKS: can't connect to GKS socket application

### ISSUE THREAD

**Issue by programistawpf** [JuliaPlots/Plots.jl#1649]
Win7 64 SP2,
Completly new instalation of Julia and GR (gksqt.exe), after clean disk. Added only (and depended) one pacakges Plots. The error is sometimes with this seme data and code ! I.e. In the same loop somtimes is sometimes no !
```
  | | |_| | | | (_| |  |  Version 0.6.2 (2017-12-13 18:08 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32
```

```
julia> using Plots
INFO: Precompiling module Plots.

julia> plot()
INFO: Precompiling module GR.
connect: No error
GKS: can't connect to GKS socket application
Did you start 'gksqt'?

GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in the state WSOP or WSAC in routine ACTIVATE_WS
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
.... many simliar lines 
KS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT

julia> 
```

**Comment by sprders** [comment]
Have you solved it now?

**Comment by mkborregaard** [comment]
I have a feeling this is a GR issue, not plots. What happens if you open a fresh session and do `using GR`? What if you try plotting with GR?

**Comment by aramirezreyes** [comment]
Have the same error on Linux64, is there any update on this?

**Comment by cmcaine** [comment]
@aramirezreyes I had a similar issue on Linux and just had to install qt4. Are you sure you've got libQtGui?

**Comment by aramirezreyes** [comment]
@cmcaine Yes, I have it. It is some weird behavior that sometimes appear but sometimes no. I will try to find the minimum conditions to reproduce it.

**Comment by heitorPB** [comment]
I have the same problem in ArchLinux. The problem is a missing library:

```
$ cd  ~/.julia/packages/GR/IVBgs/deps/gr/bin
$ ldd gksqt 
	linux-vdso.so.1 (0x00007ffffe1eb000)
	libQtGui.so.4 => not found
	libQtNetwork.so.4 => not found
	libQtCore.so.4 => not found
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f01e71b2000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f01e7023000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f01e6e9e000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f01e6e82000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f01e6cbe000)
	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f01e721c000)
```

**Comment by HaoLi111** [comment]
Similar warning show up for Julia 1.1.0
when a sequence of plots (around 1e5 plots) are saved, after some time firewall prompt shows up and ask for permission for gksqt to get online.

The plots can be saved then with a relatively slow speed.
and the warning shows:
```
GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in the state WSOP or WSAC in routine ACTIVATE_WS
connect: No error
GKS: can't connect to GKS socket application
Did you start 'gksqt'?

GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in the state WSOP or WSAC in routine ACTIVATE_WS

```
However if  a relatively smaller quantity of plots are saved (less than around 1e4 plots) then this warning would not show up

**Comment by Naba7** [comment]
I am getting the similar issue along with added errors which I have mentioned below.
```
connect: Connection refused
GKS: can't connect to GKS socket application
Did you start 'gksqt'?

GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in the state WSOP or WSAC in routine ACTIVATE_WS
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
```

**Comment by programistawpf** [comment]
My Error on win7 at this moment some time is still.Ii have only 1 version julia on my WIN7 profile , and i must first run manualy gksqt.exe . 
Paul

**Comment by SimonDanisch** [comment]
Same problem:
Julia 1.3.1, Ubuntu 19.10:
```
(v1.3) pkg> st Plots
    Status `~/.julia/environments/v1.3/Project.toml`
  [28b8d3ca] GR v0.46.0
  [91a5bcdd] Plots v0.29.2
```
Tried to update to GR 0.47.0, but plots didn't like that (as in didn't allow to update because of version bounds)
@jheinen, any idea?

**Comment by programistawpf** [comment]
Yes, a seen it. 
For WIN : gksqt.exe a set shortcut  to this into \Startup (AUTOSTART)
localization in RUN: shell:common startup
It HELP :) 
Paul

**Comment by SimonDanisch** [comment]
Seems one needs to install additional dependencies on linux:
https://gr-framework.org/julia.html#installation
Would be nice to have these instructions in an error if those libraries aren't found!

**Comment by programistawpf** [comment]
MAybe this test some help?
```
[ Info: Testing plot: Isosurface plot
[ Info: Testing plot: Shade points
[ Info: Testing plot: Discrete plot
ERROR: LoadError: IOError: unlink: resource busy or locked (EBUSY)
Stacktrace:
 [1] uv_error at .\libuv.jl:85 [inlined]
 [2] unlink(::String) at .\file.jl:743
 [3] #rm#9(::Bool, ::Bool, ::Function, ::String) at .\file.jl:253
 [4] rm at .\file.jl:245 [inlined]
 [5] macro expansion at .\logging.jl:311 [inlined]
 [6] basic_tests() at C:\Users\PC\.julia\packages\GR\f1Iqi\test\runtests.jl:222
 [7] top-level scope at util.jl:156
 [8] include at .\boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at .\loading.jl:1038
 [10] include(::Module, ::String) at .\sysimg.jl:29
 [11] include(::String) at .\client.jl:398
 [12] top-level scope at none:0
in expression starting at C:\Users\PC\.julia\packages\GR\f1Iqi\test\runtests.jl:234
ERROR: Package GR errored during testing
```
Win7 64, Julia:
```
 Version 0.7.0 (2018-08-08 06:46 UTC)
 Official http://julialang.org/ release
 x86_64-w64-mingw32
```

**Comment by jheinen** [comment]
I'd recommend to switch to a 1.x version. With GR we already support 2 dozen platforms, but there are personal and technical (CI/CD) limits somewhere.

**Comment by programistawpf** [comment]
At Julia1_0_5 the same story :)

**Comment by jheinen** [comment]
Is `gksqt.exe` in `readdir(joinpath(ENV["GRDIR"],"bin"))` ?

**Comment by programistawpf** [comment]
julia> readdir(joinpath(ENV["GRDIR"],"bin"))
ERROR: KeyError: key "GRDIR" not found
Stacktrace:
 [1] (::getfield(Base, Symbol("##423#424")))(::String) at .\env.jl:76
 [2] access_env(::getfield(Base, Symbol("##423#424")), ::String) at .\env.jl:14
 [3] getindex(::Base.EnvDict, ::String) at .\env.jl:76
 [4] top-level scope at none:0

julia>

**Comment by jheinen** [comment]
```
using GR
readdir(joinpath(ENV["GRDIR"],"bin"))
```

**Comment by jheinen** [comment]
If you don't have a `GRDIR` environment, you can use:
```
using GR
readdir(joinpath(dirname(pathof(GR)),"..","deps","gr","bin"))
```

**Comment by programistawpf** [comment]
33 items:

```
julia> using GR

julia> readdir(joinpath(dirname(pathof(GR)),"..","deps","gr","bin"))
33-element Array{String,1}:
 "cairoplugin.dll"
 "d3dcompiler_47.dll"
 "gksqt.exe"
 "gksqt.exe - skrót.lnk"
 "glplugin.dll"
 "gsplugin.dll"
 "gtkplugin.dll"
 "libgcc_s_dw2-1.dll"
 "libgcc_s_seh-1.dll"
 "libgcc_s_sjlj-1.dll"
 "libgks.dll"
 "libgks.lib"
 "libGLESv2.dll"
 "libGLESv2d.dll"
 "libgr.dll"
 "libgr.lib"
 "libGR3.dll"
 "libstdc++-6.dll"
 "libwinpthread-1.dll"
 "pgfplugin.dll"
 "platforms"
 "Qt5Core.dll"
 "Qt5Gui.dll"
 "Qt5Network.dll"
 "qt5plugin.dll"
 "Qt5Widgets.dll"
 "qtplugin.dll"
 "svgplugin.dll"
 "videoplugin.dll"
 "wmfplugin.dll"
 "wxplugin.dll"
 "x11plugin.dll"
 "zmqplugin.dll"

```

**Comment by jheinen** [comment]
The only strange thing is the file `gksqt.exe - skrót.lnk`. Are there any UTF characters in your username? Is the user's name `skrót`?

**Comment by programistawpf** [comment]
hmmmm. .Wii make it when i made right mouese shortuct for AUTOSART   , this is in polish shortuct :/  i can remowe this. Treble was beafore it.

**Comment by programistawpf** [comment]
Now is clear:
```
julia> readdir(joinpath(dirname(pathof(GR)),"..","deps","gr","bin"))
32-element Array{String,1}:
 "cairoplugin.dll"
 "d3dcompiler_47.dll"
 "gksqt.exe"
 "glplugin.dll"
 "gsplugin.dll"
 "gtkplugin.dll"
 "libgcc_s_dw2-1.dll"
 "libgcc_s_seh-1.dll"
 "libgcc_s_sjlj-1.dll"
 "libgks.dll"
 "libgks.lib"
 "libGLESv2.dll"
 "libGLESv2d.dll"
 "libgr.dll"
 "libgr.lib"
 "libGR3.dll"
 "libstdc++-6.dll"
 "libwinpthread-1.dll"
 "pgfplugin.dll"
 "platforms"
 "Qt5Core.dll"
 "Qt5Gui.dll"
 "Qt5Network.dll"
 "qt5plugin.dll"
 "Qt5Widgets.dll"
 "qtplugin.dll"
 "svgplugin.dll"
 "videoplugin.dll"
 "wmfplugin.dll"
 "wxplugin.dll"
 "x11plugin.dll"
 "zmqplugin.dll"

```

**Comment by programistawpf** [comment]
My username is PC

**Comment by jheinen** [comment]
I'm aware that there are problems with UTF characters in files used by GR.
As this doesn't apply to your system, I currently have no idea why `gksqt` is not started automatically. It works fine in several Windows boxes at our site, most of them Windows 10.

You could try to set an environment variable `GKS_QT` to the absolute path of you `gksqt.exe` binary and see what happens.

In your case something like:
```
set GKS_QT=C:\Users\PC\.julia\packages\GR\f1Iqi\deps\gr\bin\gksqt.exe
```

**Comment by mkborregaard** [comment]
I agree with Simon that a clearer error message with a suggestion for a fix would go really far here.

**Comment by programistawpf** [comment]
W dniu 2020-02-28 o 11:29, Josef Heinen pisze:
After:
> |set GKS_QT=C:\Users\PC\.julia\packages\GR\f1Iqi\deps\gr\bin\gksqt.exe|

looks nice...

Tbanks

Paul

**Comment by jheinen** [comment]
Good new - but I still would like to understand why it doesn't work without this setting ...

**Comment by programistawpf** [comment]
The WIN does not behave unequivocally in relation to GR. Look like he 
has moments when he ignores it. I tested this purely random case :/

Paul
W dniu 2020-02-28 o 14:51, Josef Heinen pisze:

**Comment by jheinen** [comment]
If `GKS_QT` is set **before** you start Julia (and later `gksqt`) then it really doesn't make sense !?

**Comment by jheinen** [comment]
> I agree with Simon that a clearer error message with a suggestion for a fix would go really far here.

There's now a check during the build process in GR [master](https://github.com/jheinen/GR.jl/commit/a4770e1f54a631f78d98f41726e248e13d4247d0).

**Comment by programistawpf** [comment]
Only 1 time set, I did,
After this gksqt.exe was not started manulay
Paul
W dniu 2020-02-28 o 15:29, Josef Heinen pisze:

**Comment by aminya** [comment]
Any solution for this? I am trying to write a SnoopCompile bot for Plots, but it gives me this error. This does not happen inside the normal GitHub Action CI that I made (#2745).

https://github.com/aminya/Plots.jl/runs/724208660?check_suite_focus=true#step:8:1

**Comment by jheinen** [comment]
Did you set
```
ENV["PLOTS_TEST"] = "true"
ENV["GKSwstype"] = "100"
```
as in the `Plots.jl` test set?

**Comment by jheinen** [comment]
Did you set
```
ENV["PLOTS_TEST"] = "true"
ENV["GKSwstype"] = "100"
```
as in the `Plots.jl` test set?

**Comment by aminya** [comment]
> Did you set
> 
> ```
> ENV["PLOTS_TEST"] = "true"
> ENV["GKSwstype"] = "100"
> ```
> 
> as in the `Plots.jl` test set?

Do I need to do this from inside Julia? I added it as a job environment variable and my job is running for 45min without any logs being printed.
https://github.com/aminya/Plots.jl/runs/724427199?check_suite_focus=true

**Comment by Terkwood** [comment]
Eventually got this working with popOS / Julia 1.5.1 by installing qt5

```sh
apt install -y qtbase5-dev-tools  qt5-default  # maybe only qt5-default is necessary
```

**Comment by bgoodr** [comment]
> Eventually got this working with popOS / Julia 1.5.1 by installing qt5
> 
> ```shell
> apt install -y qtbase5-dev-tools  qt5-default  # maybe only qt5-default is necessary
> ```

This worked for me, as I ran into the exact same situation on Ubuntu 20.04.  So thank you!

**Comment by RajGhugare19** [comment]
@Terkwood Thank you! I am using ubuntu 20.04, and was trying was ReinforcementLearningEnvironments. I was facing the same problem,this worked out for me

**Comment by ashwani-rathee** [comment]
@Terkwood I was facing the same issue, this worked for me too, in Ubuntu 20.04. thank you!

**Comment by benneti** [comment]
I have the same problem on nixos with julia version 1.5.3

```
julia> using Plots # Plots v1.10.4
[ Info: Precompiling Plots [91a5bcdd-55d7-5caf-9e0b-520d859cae80]

julia> plot(sin)
env: ‘/home/benneti/.julia/artifacts/75305428f7454d8c7304e183ae65ca13bc0236c1/bin/gksqt’: No such file or directory
connect: Connection refused
GKS: can't connect to GKS socket application

GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in the state WSOP or WSAC in routine ACTIVATE_WS
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine TEXT
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine POLYLINE

shell> ldd /home/benneti/.julia/artifacts/75305428f7454d8c7304e183ae65ca13bc0236c1/bin/gksqt
	linux-vdso.so.1 (0x00007ffd990f7000)
	libQt5Widgets.so.5 => /nix/store/b33sv42xxspsnhfspv1n04wi0wbycq7k-qtbase-5.15.2/lib/libQt5Widgets.so.5 (0x00007fc73b870000)
	libQt5Network.so.5 => /nix/store/b33sv42xxspsnhfspv1n04wi0wbycq7k-qtbase-5.15.2/lib/libQt5Network.so.5 (0x00007fc73b6ea000)
	libpthread.so.0 => /nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib/libpthread.so.0 (0x00007fc73b6c9000)
	libfreetype.so.6 => /nix/store/34siz8mka96767blk28xq3vvxpcyjvv2-freetype-2.10.4/lib/libfreetype.so.6 (0x00007fc73b605000)
	libpng16.so.16 => /nix/store/w9myg0z6v7hsv1qnxcps69gxi9m7lxv7-libpng-apng-1.6.37/lib/libpng16.so.16 (0x00007fc73b5cb000)
	libz.so.1 => /nix/store/lnjmyxgqq27dsmnc2sxwa7hn448bzc4k-zlib-1.2.11/lib/libz.so.1 (0x00007fc73b5ac000)
	libdl.so.2 => /nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib/libdl.so.2 (0x00007fc73b5a7000)
	libQt5Gui.so.5 => /nix/store/b33sv42xxspsnhfspv1n04wi0wbycq7k-qtbase-5.15.2/lib/libQt5Gui.so.5 (0x00007fc73af44000)
	libQt5Core.so.5 => /nix/store/b33sv42xxspsnhfspv1n04wi0wbycq7k-qtbase-5.15.2/lib/libQt5Core.so.5 (0x00007fc73a98c000)
	libstdc++.so.6 => /nix/store/q3y6lqylj69ya2jr6b8v4jgcp11j8ldk-gfortran-9.3.0-lib/lib/libstdc++.so.6 (0x00007fc73a7ab000)
	libm.so.6 => /nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib/libm.so.6 (0x00007fc73a668000)
	libgcc_s.so.1 => /nix/store/q3y6lqylj69ya2jr6b8v4jgcp11j8ldk-gfortran-9.3.0-lib/lib/libgcc_s.so.1 (0x00007fc73a64c000)
	libc.so.6 => /nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib/libc.so.6 (0x00007fc73a48b000)
	libGL.so.1 => /nix/store/821fd1mbgf8036kmns0rdyaycgg3bc8y-libGL-1.3.2/lib/libGL.so.1 (0x00007fc73a3fd000)
	libgssapi_krb5.so.2 => /nix/store/kk4qf5d99x6kpcjhdlkki4yifzgmxmlm-libkrb5-1.18/lib/libgssapi_krb5.so.2 (0x00007fc73a3aa000)
	libssl.so.1.1 => /nix/store/wmhdjm0x0n4ffqh908gkhah3zr1d29fd-openssl-1.1.1i/lib/libssl.so.1.1 (0x00007fc73a312000)
	libcrypto.so.1.1 => /nix/store/wmhdjm0x0n4ffqh908gkhah3zr1d29fd-openssl-1.1.1i/lib/libcrypto.so.1.1 (0x00007fc73a025000)
	/lib64/ld-linux-x86-64.so.2 => /nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib64/ld-linux-x86-64.so.2 (0x00007fc73bf0d000)
	libbz2.so.1 => /nix/store/hc96d6rsfcbagxzbn8xwsm32bzizsj69-bzip2-1.0.6.0.1/lib/libbz2.so.1 (0x00007fc73a012000)
	libharfbuzz.so.0 => /nix/store/wbdlg8p4ikn3xa0g5qcbbxzrx5rc7xgg-harfbuzz-2.7.2/lib/libharfbuzz.so.0 (0x00007fc739f2e000)
	libicui18n.so.67 => /nix/store/9vpwg9lgnjp9a1fqzcsxcmq4c1r5vigp-icu4c-67.1/lib/libicui18n.so.67 (0x00007fc739c16000)
	libicuuc.so.67 => /nix/store/9vpwg9lgnjp9a1fqzcsxcmq4c1r5vigp-icu4c-67.1/lib/libicuuc.so.67 (0x00007fc739a28000)
	libicudata.so.67 => /nix/store/9vpwg9lgnjp9a1fqzcsxcmq4c1r5vigp-icu4c-67.1/lib/libicudata.so.67 (0x00007fc737f11000)
	libpcre2-16.so.0 => /nix/store/3ik2szvrw86yy6mg4wq1yiaswv9jyxdk-pcre2-10.36/lib/libpcre2-16.so.0 (0x00007fc737e85000)
	libgthread-2.0.so.0 => /nix/store/hnk0s558k2h4v8abjzagagflvw1l0cb2-glib-2.66.4/lib/libgthread-2.0.so.0 (0x00007fc737e80000)
	libglib-2.0.so.0 => /nix/store/hnk0s558k2h4v8abjzagagflvw1l0cb2-glib-2.66.4/lib/libglib-2.0.so.0 (0x00007fc737d4c000)
	libGLX.so.0 => /nix/store/q6fza5n00mfk12y3lz3lwcas4gswyaxx-libglvnd-1.3.2/lib/libGLX.so.0 (0x00007fc737d17000)
	libX11.so.6 => /nix/store/6y53vmypx0lmd0r43sa8qsd026yqfsmz-libX11-1.7.0/lib/libX11.so.6 (0x00007fc737bd2000)
	libXext.so.6 => /nix/store/dm9k4ryj2dykk32z560a0mfnlf9c3pb8-libXext-1.3.4/lib/libXext.so.6 (0x00007fc737bbd000)
	libGLdispatch.so.0 => /nix/store/q6fza5n00mfk12y3lz3lwcas4gswyaxx-libglvnd-1.3.2/lib/libGLdispatch.so.0 (0x00007fc737b04000)
	libkrb5.so.3 => /nix/store/kk4qf5d99x6kpcjhdlkki4yifzgmxmlm-libkrb5-1.18/lib/libkrb5.so.3 (0x00007fc737a28000)
	libk5crypto.so.3 => /nix/store/kk4qf5d99x6kpcjhdlkki4yifzgmxmlm-libkrb5-1.18/lib/libk5crypto.so.3 (0x00007fc7379f8000)
	libcom_err.so.3 => /nix/store/kk4qf5d99x6kpcjhdlkki4yifzgmxmlm-libkrb5-1.18/lib/libcom_err.so.3 (0x00007fc7379f2000)
	libkrb5support.so.0 => /nix/store/kk4qf5d99x6kpcjhdlkki4yifzgmxmlm-libkrb5-1.18/lib/libkrb5support.so.0 (0x00007fc7379e2000)
	libkeyutils.so.1 => /nix/store/z5ai50bfi7raz4wsb0r6ijjm6mxlg266-keyutils-1.6.3-lib/lib/libkeyutils.so.1 (0x00007fc7379db000)
	libresolv.so.2 => /nix/store/gafigwfaimlziam6qhw1m8dz4h952g1n-glibc-2.32-35/lib/libresolv.so.2 (0x00007fc7379c1000)
	libgraphite2.so.3 => /nix/store/z2gfpvs7a371d81liv7i899yj0ypc6ig-graphite2-1.3.14/lib/libgraphite2.so.3 (0x00007fc737997000)
	libpcre.so.1 => /nix/store/zs95g44vip9834nlp6qqci06mysblkq2-pcre-8.44/lib/libpcre.so.1 (0x00007fc73791e000)
	libxcb.so.1 => /nix/store/0k01pc905yqyyg0fc8c9p400vizwwcdj-libxcb-1.14/lib/libxcb.so.1 (0x00007fc7378f2000)
	libXau.so.6 => /nix/store/0xylz23kx7bsz0gv4dhg5xwv301gi7yi-libXau-1.0.9/lib/libXau.so.6 (0x00007fc7378eb000)
	libXdmcp.so.6 => /nix/store/cq898007g7a86w5y0bp6dwjivq64i474-libXdmcp-1.1.3/lib/libXdmcp.so.6 (0x00007fc7378e3000)
```

but using GR, plotting and saving works just fine.

**Comment by ErikPillon** [comment]
The same happened to me with my `1.6.0 julia` on macOS BigSur version 11.2.3.

I solved the problem by simply adding the `gr()` backend to my script.

Hope this helps!

**Comment by YutaroIshihama** [comment]
The same happened to me. Even worse, adding the `gr()` backend apparently to my script did **not resolve the problem**.
My OS is Win10.  With seeing `C:\Users\XXX\.julia\packages\GR` (`XXX` is my user name), there are three folders. The folder
having the oldest time stamp includes `gksqt.exe`, but the other new folders do not. Am I failing to set a path for `gksqt.exe`?
Besides, I have no idea how to set `GKS_QT`. My trial using `Julia`'s shell mode failed.

**Comment by jheinen** [comment]
Please update GR (to version 0.57.3) and the problems should be fixed.

**Comment by spasmsagain** [comment]
Hi

I've installed Julia 1.6.0 and Plots on a Window 10 64 bit machine.

I'm getting errors messages:
```
connect: No error
GKS: can't connect to GKS socket application

GKS: Open failed in routine OPEN_WS
GKS: GKS not in proper state. GKS must be either in the state WSOP or WSAC in routine ACTIVATE_WS
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
GKS: GKS not in proper state. GKS must be either in the state WSAC or SGOP in routine FILLAREA
```
etc.

Further I've tried
 add 
```
C:\Users\Username\.julia\artifacts\c34a53538bc6c87c8b376c3f6d0d3248830003bc\bin
````
to the path

and setting GKS_QT
```
set GKS_QT=C:\Users\Username\.julia\artifacts\7cf489c280ce03e932b09698ef4d949827fd9c55\bin\gksqt.exe
```
but that only adds to problem as I now have 
![gksqt error](https://user-images.githubusercontent.com/43171192/116501390-ff610600-a8f3-11eb-800b-a3f1ec1e4786.png)

as a new a new error message

Question:
Why is the QT5 gksqt.exe binary installed in a different directory to the QT5.DLLS

I don't really to be adding weird path names to the PATH variables and having to set environmental variables with strange path names for this to work.

Reiterating this is on a Windows 10 laptop.

Regards

**Comment by sajad-gh** [comment]
Guys. Please Help. I have the same error trying to use plot in my loptop. What can do to solve my issue

**Comment by jheinen** [comment]
What is your OS?

**Comment by sajad-gh** [comment]
Windows 10 Enterprise- 64bit
I installed the same atoms and Julia on my friend's loptop which has windows 7 . it works and I did my project but I want it on my loptop

**Comment by joaquinmura** [comment]
Same issue reported [here](https://stackoverflow.com/editing-help). At least for me on Windows 10, the problem was that the antivirus blocked `gksqt.exe.` Unblocking makes it run ok.

**Comment by jheinen** [comment]
The problem should be fixed with the latest GR version.
